apikey-service/src/main/java/eu/europeana/apikey/keycloak/KeycloakManager.java

```diff
@@ -543,15 +543,16 @@ public boolean isClientAuthorized(String apikey, KeycloakAuthenticationToken key
      *
      * @param enable enable or disable a client
      * @param clientId client identifier
+     * @param apikeyUpdate update information which will be used in case of enabling
      * @param securityContext security context with access token
      * @throws ApikeyException when client not found in Keycloak or update failed
      */
-    public void enableClient(boolean enable, String clientId, KeycloakSecurityContext securityContext) throws ApikeyException {
+    public void enableClient(boolean enable, String clientId, ApikeyDetails apikeyUpdate, KeycloakSecurityContext securityContext) throws ApikeyException {
         ClientRepresentation clientRepresentation = getClientRepresentation(clientId, securityContext);
         if (clientRepresentation == null) {
             throw new ApikeyException(HttpStatus.SC_NOT_FOUND);
         }
-        if (clientRepresentation.isEnabled() != enable) {
+        if (clientRepresentation.isEnabled() != enable && !enable) {
```

Comment:
This looks very weird. What exactly are you checking here?

Reply:
Actually it's mentioned in the method description: "If the status of the client wouldn't change after the update the call to Keycloak is skipped". The thing is that if you want to enable a client using API but it had already been enabled in Keycloak console there is no point to execute the request. The part "&& !enable" is here just to be sure that only disable is considered because that's the scope of this task.